packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/open-telemetry.ts

```diff
@@ -0,0 +1,243 @@
+            - ecs.task.storage.write_bytes
+    metricstransform:
+      transforms:
+        - metric_name: ecs.task.memory.utilized
+          action: update
+          new_name: MemoryUtilized
+        - metric_name: ecs.task.memory.reserved
+          action: update
+          new_name: MemoryReserved
+        - metric_name: ecs.task.cpu.utilized
+          action: update
+          new_name: CpuUtilized
+        - metric_name: ecs.task.cpu.reserved
+          action: update
+          new_name: CpuReserved
+        - metric_name: ecs.task.network.rate.rx
+          action: update
+          new_name: NetworkRxBytes
+        - metric_name: ecs.task.network.rate.tx
+          action: update
+          new_name: NetworkTxBytes
+        - metric_name: ecs.task.storage.read_bytes
+          action: update
+          new_name: StorageReadBytes
+        - metric_name: ecs.task.storage.write_bytes
+          action: update
+          new_name: StorageWriteBytes
+    resource:
+      attributes:
+        - key: ClusterName
+          from_attribute: aws.ecs.cluster.name
+          action: insert
+        - key: aws.ecs.cluster.name
+          action: delete
+        - key: ServiceName
+          from_attribute: aws.ecs.service.name
+          action: insert
+        - key: ServiceName
+          action: update
+          value: ${props.serviceName}
+        - key: aws.ecs.service.name
+          action: delete
+        - key: TaskId
+          from_attribute: aws.ecs.task.id
+          action: insert
+        - key: aws.ecs.task.id
+          action: delete
+        - key: TaskDefinitionFamily
+          from_attribute: aws.ecs.task.family
+          action: insert
+        - key: aws.ecs.task.family
+          action: delete
+  exporters:
+    awsxray:
+    awsemf/application:
+      log_group_name: 'aws/ecs/containerinsights/{ClusterName}/application/metrics'
+      log_stream_name: '{TaskId}'
+    awsemf/performance:
+      namespace: ECS/ContainerInsights
+      log_group_name: '/aws/ecs/containerinsights/{ClusterName}/performance'
+      log_stream_name: '{TaskId}'
+      resource_to_telemetry_conversion:
+        enabled: true
+      dimension_rollup_option: NoDimensionRollup
+      metric_declarations:
+        dimensions: [ [ ClusterName ], [ ClusterName, TaskDefinitionFamily ], [ ClusterName, ServiceName ] ]
+        metric_name_selectors: [ . ]
+  service:
+    pipelines:
+      traces:
+        receivers: [otlp,awsxray]
+        processors: [batch/traces]
+        exporters: [awsxray]
+      metrics/application:
+        receivers: [otlp]
+        processors: [batch/metrics]
+        exporters: [awsemf/application]
+      metrics/performance:
+        receivers: [awsecscontainermetrics ]
+        processors: [filter, metricstransform, resource]
+        exporters: [ awsemf/performance ]
+
+    extensions: [health_check]
+  `;
+}
+
+/**
+ * Settings for the main application container
+ */
+export interface OpenTelemetryProps {
+
+}
```

Comment:
Do you want to add props here and where do you actually use this Props object? Since this doesn't currently have any fields nor does it extend anything, what future props can you see being added here?